### DISCUSSION TITLE
Generated migrations with null safety

### DISCUSSION THREAD

**Discussion by MiguelSOliveira** [simolus3/drift#1185] [Q&A] [answered]
Hi, I'm trying to migrate my app to null safety but I'm getting the following error:

```
Bad state: Error: package has unmigrated dependencies.

Before migrating your package, we recommend ensuring that every library it
imports (either directly or indirectly) has been migrated to null safety, so
that you will be able to run your unit tests in sound null checking mode.  You
are currently importing the following non-null-safe libraries:

  package:X/database/test/generated_migrations/schema_v10.dart
  package:X/database/test/generated_migrations/schema_v11.dart
  package:X/database/test/generated_migrations/schema_v12.dart
  package:X/database/test/generated_migrations/schema_v13.dart
  package:X/database/test/generated_migrations/schema_v14.dart
  package:X/database/test/generated_migrations/schema_v7.dart
  package:X/database/test/generated_migrations/schema_v8.dart
  package:X/database/test/generated_migrations/schema_v9.dart

Please upgrade the packages containing these libraries to null safe versions
before continuing.  To see what null safe package versions are available, run
the following command: `dart pub outdated --mode=null-safety`.

To skip this check and try to migrate anyway, re-run with the flag
`--skip-import-check`.
```

I think it's because each generated migration has the following:

```
// GENERATED CODE, DO NOT EDIT BY HAND.
//@dart=2.9
import 'package:moor/moor.dart';
```

Is it possible to generate these files again with null safety? I have each of these migrations' json schema files.

Thank you!

**Comment by simolus3** [comment, marked as answer]
You can add the `--null-safety` flag to `moor_generator schema generate` to generate null-safe code. I'll take a look at generating null safe code by default if the surrounding package is opted in.

**Reply by kuhnroyal** [reply]
I am using `LibraryElement.isNonNullableByDefault` to detect this.

**Reply by simolus3** [reply]
`moor_generator` also does that when generating code, but here it is a bit more complicated because we're not generating a part file that would belong to a library. In 2fe81515089fdf3d0cbcdcbdaad6156a71b627d8, I instead look at the `.dart_tool/package_config.json` file to determine whether the surrounding package is opted into null safety.
